### ISSUE TITLE
Ringing - From: null, To: null, Direction: CallDirection.outgoing

### ISSUE THREAD

**Issue by preprincipletech** [cybex-dev/twilio_voice#57]
: Ringing - From: null, To: null, Direction: CallDirection.outgoing
I/flutter (11315): voip-onCallStateChanged CallEvent.ringing
I/flutter (11315): ringing
D/TwilioVoicePlugin(11315): Connect failure
E/TwilioVoicePlugin(11315): Call Error: 31603, Decline
I/flutter (11315): Call Error: 31603, Decline

**Comment by Erchil66** [comment]
We do have same also i get my access token own our own server and token for fcm for the device but still no luck same as the error you give. And we don's use TWIML just on putting in phone servers.

**Comment by diegogarciar** [comment]
Please review your Twilio logs to ensure you don't have any errors/call fails from Flutter.

**Comment by cybex-dev** [comment]
@Erchil66 see #56 - this could be useful

**Comment by cybex-dev** [comment]
@preprincipletech see #56, could be useful

**Comment by agent515** [comment]
Hi @cybex-dev .

I'm seeing similar logs despite call getting connected. `From` and `To` are also not available on the caller side as you confirmed [here](https://github.com/diegogarciar/twilio_voice/issues/105#issuecomment-1260941399). How to resolve this as the steps in #56 are unclear to me.

Platform: Android

Client code:
```dart
  bool? result = await _instance.call.place(
    from: "client:Alice",
    to: "Bob",
    extraOptions: null,
  );
```

`makeCall` is the name of the function registered as callback for TwilioML app. The function receives following request body.
```json
{
  AccountSid: <ACCOUNT_SID>,
  ApiVersion: '2010-04-01',
  ApplicationSid: <ACCOUNT_SID>,
  CallSid: <CALL_SID>,
  CallStatus: 'ringing',
  Called: '',
  Caller: 'client:Alice',
  Direction: 'inbound',
  From: 'client:Alice',
  To: 'Bob'
}
```

Dialing call inside function.
```js
 const dial = voiceResponse.dial({
        callerId: request.body.From,
        answerOnBridge: true,
      });
```

VoiceResponse in XML
```xml
<?xml version="1.0" encoding="UTF-8"?><Response><Dial callerId="client:Alice" answerOnBridge="true"><Client>Bob</Client></Dial></Response>
```

**Comment by cybex-dev** [comment]
@agent515 

Did you resolve this problem? `31603` does ring a bell, but I do not recall the exact cause though I'm reminded of a client not being registered/active or something thereabout.

**Comment by agent515** [comment]
Hi @cybex-dev. I am able to receive correct values for `From` and `To` on callee side. But `CallDirection` is still `CallDirection.outgoing`. But not sure how I resolved this because I noticed this working just now (only on the callee side). Maybe it started working after taking latest pull from the master.

Caller:
`Ringing - From: null, To: null, Direction: CallDirection.outgoing`

Callee:
`Ringing - From: Alice, To: Bob, Direction: CallDirection.outgoing`
